graphs/eacs/src/test/kotlin/com/anaplan/engineering/azuki/graphs/eacs/BEH5.kt

```diff
@@ -0,0 +1,126 @@
+package com.anaplan.engineering.azuki.graphs.eacs
+
+import com.anaplan.engineering.azuki.core.runner.Eac
+import com.anaplan.engineering.azuki.core.system.BEH
+import com.anaplan.engineering.azuki.graphs.adapter.api.GraphBehaviours
+import com.anaplan.engineering.azuki.graphs.adapter.api.GraphFunctionalElements
+import com.anaplan.engineering.azuki.graphs.dsl.GraphScenario
+import com.anaplan.engineering.azuki.graphs.graphA
+
+@BEH(GraphBehaviours.GetShortestPath, GraphFunctionalElements.GraphFunction, """
+
+""")
+class BEH5 : GraphScenario() {
+
+    @Eac("""
+        Given a graph that contains at least path from a beginning vertex to an ending vertex,
+        the graph will have a shortest path between these vertices
+    """)
+    fun onePath() {
+        given {
+            thereIsAnUndirectedGraph(graphA) {
+                edge("a", "b")
+            }
+        }
+        then {
+            hasShortestPath(graphA, "a", "b")
+        }
+    }
+
+    @Eac("""
+        Given a graph that contains more then one path between two vertices,
+        there is a shortest path
+    """, """
+        If two paths have the same number of vertices and are both the shortest only
+        the first path defined will be recognised as the shortest path.
```

Comment:
🚫 I have a strong feeling that this is implementation dependent. At the very least, there is no guarantee that the declaration will have any specific order, so if the order truly was important you'd need to add the edges one by one in the `whenever` block. I would say something like
```suggestion
        If there are multiple paths with minimum length between the given endpoints, 
        the choice of which one is returned is implementation dependent.
```